pkg/issuer/acme/setup.go

```diff
@@ -97,6 +98,55 @@ func (a *Acme) Setup(ctx context.Context) (issuer.SetupResponse, error) {
 		return issuer.SetupResponse{Requeue: true}, err
 	}
 
+	// TODO: perform a complex check to determine whether we need to verify
+	// the existing registration with the ACME server.
+	// This should take into account the ACME server URL, as well as a checksum
+	// of the private key's contents.
+	// Alternatively, we could add 'observed generation' fields here, tracking
+	// the most recent copy of the Issuer and Secret resource we have checked
+	// already.
+
+	rawServerURL := a.issuer.GetSpec().ACME.Server
+	parsedServerURL, err := url.Parse(a.issuer.GetStatus().ACMEStatus().URI)
+	if err != nil {
+		a.Recorder.Eventf(a.issuer, v1.EventTypeWarning, "InvalidURL",
+			"Failed to parse existing ACME server URI %q - "+
+				"re-validating ACME server registration: %v", rawServerURL, err)
+		// absorb errors as retrying will not help resolve this error
+		return issuer.SetupResponse{}, nil
+	}
+
+	rawAccountURL := a.issuer.GetStatus().ACMEStatus().URI
+	parsedAccountURL, err := url.Parse(a.issuer.GetStatus().ACMEStatus().URI)
+	if err != nil {
+		a.Recorder.Eventf(a.issuer, v1.EventTypeWarning, "InvalidURL",
+			"Failed to parse existing ACME account URI %q - "+
+				"re-validating ACME account registration: %v", rawAccountURL, err)
+		// absorb errors as retrying will not help resolve this error
+		return issuer.SetupResponse{}, nil
+	}
+
+	hasReadyCondition := a.issuer.HasCondition(v1alpha1.IssuerCondition{
+		Type:   v1alpha1.IssuerConditionReady,
+		Status: v1alpha1.ConditionTrue,
+	})
+
+	// If the Host components of the server URL and the account URL match, then
+	// we skip re-checking the account status to save excess calls to the
+	// ACME api.
+	if hasReadyCondition &&
+		a.issuer.GetStatus().ACMEStatus().URI != "" &&
+		parsedAccountURL.Host == parsedServerURL.Host {
```

Comment:
We now skip rechecking if:

* the issuer is already marked Ready (this is our 'standard' status condition for checking readiness of issuer resources)
* the account URL is set
* the account URL and the server URL host components match